### ISSUE TITLE
amp-twitter fails with dynamic data-tweetid

### ISSUE THREAD

**Issue by kedar9444** [ampproject/amphtml#17953]
I am trying to integrate [these ](https://ampbyexample.com/playground/#url=https%3A%2F%2Fampbyexample.com%2Fcomponents%2Famp-twitter%2Fsource%2F)awesomely 😍 written AMP web components in my [website ](https://angularpoc.com/about/platform)[angular content/resource platform] but facing issue in it.
the issue i am facing is :
It works perfectly fine when i use the static tweetid in amp twitter component 

`  <amp-twitter width="375" height="472"
               layout="responsive"
               data-tweetid="638793490521001985">
  </amp-twitter>`

but when i try to make dynamic it fails.

` <amp-twitter width="375" height="472" layout="responsive"
              [data-tweetid]="id">
            </amp-twitter>`

where id is variable with data. 😬

using it with Angular 6.

can you guys explain possible reason.

thanks in advance. ❤️ ❤️ ❤️  and more love to you guys, for making these integrations so simple.

**Comment by kedar9444** [comment]
this one resolved my issue.
`<amp-twitter width="375" height="472" layout="responsive" data-tweetid="{{id}}"> </amp-twitter>`

**Comment by kedar9444** [comment]
is it relevant question? anyone looking into it..?

**Comment by kevinkassimo** [comment]
Hi @kedar9444 , sorry for the late response (weekends).
From my understanding, currently `data-tweetid` is not a bindable property. Currently supported attributes are available here: https://www.ampproject.org/docs/reference/components/amp-bind

Will talk with @nainar to see if this could be a feature request.

**Comment by kedar9444** [comment]
@kevinkassimo : thanks for update, would like volunteer and contribute to these components. can you give idea where and how i can contribute to these component.

**Comment by kevinkassimo** [comment]
@kedar9444 Thanks! I would suggest checking the [contributing guides](https://github.com/ampproject/amphtml/tree/master/contributing) and walk through [codelab for creating a component](https://codelabs.developers.google.com/codelabs/creating-your-first-amp-component/index.html?index=..%2F..%2Findex#0). Also, in this special case, I would suggest checking out `mutatedAttributesCallback` (see its definition in [base-element.js](https://github.com/ampproject/amphtml/blob/master/src/base-element.js)), `amp-bind`'s source code, and things inside `3p/` folder (to implement this feature might require quite a few research since it spans across quite a few components)

**Comment by nainar** [comment]
Hi @kedar9444 

### First Timers Only
We know figuring out the process for contributing to an open source project can be intimidating, so we created this issue as a way for you to learn the ropes.  (If you feel comfortable contributing to open source projects, please leave this issue for someone else.)

### What you will need to know
The change required is here: 
https://github.com/ampproject/amphtml/blob/master/src/purifier.js#L103

See this PR for a demo: https://github.com/ampproject/amphtml/pull/16913

Please also add a test when done to make sure this behaviour doesn't regress.

### Background

### Step by step
- [ ] Claim this issue by adding a comment below.  Please only claim this bug if you plan on starting work in the next day or so.  (If you [join the AMP Project](https://goo.gl/forms/T65peVtfQfEoDWeD3) we'll be able to assign this issue to you after you've claimed it.)
- [ ] If you aren't too familiar with Git/GitHub, see the [Getting Started End-to-End Guide](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md) for [an intro to Git & GitHub,](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#intro-to-git-and-github) and [how to get a copy of the code](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#get-a-copy-of-the-amphtml-code).  You can also refer to the [Quick Start Guide](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-quick.md) for the necessary setup steps with less explanation than the End-to-End guide.
- [ ] Follow the instructions for [building AMP](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server).
- [ ] [Create a Git branch](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#create-a-git-branch) for making your changes.
- [ ] [Sign the Contributor License Agreement](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md#contributor-license-agreement) before creating a Pull Request.  (If you are contributing code on behalf of a corporation start this process as early as possible.)

- [ ] [Commit your changes](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#edit-files-and-commit-them) frequently.
- [ ] [Push your changes to GitHub](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#push-your-changes-to-your-github-fork).

- [ ] [Create a Pull Request](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#send-a-pull-request-ie-request-a-code-review).  Mention `closes Issue <this issue number>` in the description.
- [ ] [Respond to your reviewer's comments](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#respond-to-pull-request-comments) (if any).


Once approved, your changes will be merged.  **⚡⚡⚡Congrats on making your first contribution to the AMP Project!⚡⚡⚡**  You'll be able to see it [live across the web soon](https://github.com/ampproject/amphtml/blob/master/contributing/release-schedule.md)!

Thanks, and we hope to see more contributions from you soon.

### Questions?


If you have questions ask in this issue or on your Pull Request (if you've created one) or see the [How to get help](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#how-to-get-help) section of the Getting Started guide.

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile.  Do you have any updates?

**Comment by seergiioo6** [comment]
Could I take this one?
